src/foam/demos/examples/services.fbe

```diff
@@ -0,0 +1,436 @@
+  
+  skeleton: true,
+  client: true,
+  
+  methods: [
+    {
+      name: 'createTask',
+      async: true,
+      type: 'String',  
+      args: 'String title, String description',
+      javaCode: `
+        String taskId = "task-" + System.currentTimeMillis();
+        Task task = Task.create()
+          .setId(taskId)
+          .setTitle(title)
+          .setDescription(description)
+          .setCreated(new java.util.Date());
+        
+        // Store task (simplified - use DAO in real implementation)
+        getTaskMap().put(taskId, task);
+        
+        System.out.println("Task created: " + taskId);
+        return taskId;
+      `
+    },
+    {
+      name: 'getTask',
+      async: true,
+      type: 'foam.demo.service.Task',
+      args: 'String taskId',
+      javaCode: `
+        Task task = (Task) getTaskMap().get(taskId);
+        System.out.println("Task retrieved: " + taskId + " " + (task != null ? "found" : "not found"));
+        return task;
+      `
+    },
+    {
+      name: 'deleteTask',
+      async: true, 
+      type: 'boolean',
+      args: 'String taskId',
+      javaCode: `
+        boolean existed = getTaskMap().containsKey(taskId);
+        getTaskMap().remove(taskId);
+        System.out.println("Task deleted: " + taskId + " " + (existed ? "success" : "not found"));
+        return existed;
+      `
+    },
+    {
+      name: 'getTaskMap',
+      type: 'java.util.Map',
+      javaCode: `
+        // In real implementation, use DAO or database
+        if ( taskMap_ == null ) {
+          taskMap_ = new java.util.concurrent.ConcurrentHashMap<>();
+        }
+        return taskMap_;
+      `
+    }
+  ],
+  
+  properties: [
+    {
+      class: 'Object',
+      name: 'taskMap',
+      javaType: 'java.util.Map<String, Task>',
+      visibility: 'HIDDEN'
+    }
+  ]
+});
+
+// Define Task model
+foam.CLASS({
+  package: 'foam.demo.service',
+  name: 'Task',
+  
+  properties: [
+    { class: 'String', name: 'id' },
+    { class: 'String', name: 'title' },
+    { class: 'String', name: 'description' },
+    { class: 'DateTime', name: 'created' }
+  ]
+});
+
+log('TaskService interface with default Java implementation defined');
+
+##  Separate Java Implementation
+
+For more complex services, create a separate Java file implementing the interface:
+--
+// ServerTaskService.java
+/*
+package foam.demo.service;
+
+import foam.core.ContextAwareSupport;
+import foam.dao.DAO;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.Map;
+
+public class ServerTaskService extends ContextAwareSupport implements TaskService {
+  
+  private final Map<String, Task> taskMap = new ConcurrentHashMap<>();
+  
+  @Override
+  public String createTask(String title, String description) {
+    String taskId = "task-" + System.currentTimeMillis();
+    Task task = (Task) Task.create(getX())
+      .setId(taskId)
+      .setTitle(title)
+      .setDescription(description)
+      .setCreated(new java.util.Date());
```

Comment:
So for best practice, the Task model should implement 'foam.core.auth.CreatedAware'. 
EasyDAO will automatically wrap in a CreatedAwareDAO.